### PR TITLE
EP: Introduce RestoreBatch + Fix BackupBatch Issue

### PR DESCRIPTION
**Pull request by hossainemruz**
xref: 
- https://github.com/stashed/stash/issues/1053
- https://github.com/stashed/stash/issues/1054
- https://github.com/stashed/stash/issues/1039
- https://github.com/stashed/stash/issues/1041

### CODE REVIEW

File: keps/restore-batch.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+      - /source/data
+```
+
+The `.status` section of the `RestoreBatch` should be similar to the `.status` section of the `BackupBatch`.
+
+## Backup and restore order for `BackupBatch` and `RestoreBatch`
+
+The underlying `restic` tool can backup and restore in parallel to and from a single repository for different `hosts`. So, the members of the `BackupBatch` or `RestoreBatch` can be backed up / restored in parallel. However, there might be a case where the user may need an order guarantee for the members. In this case, we can do the followings:
+
+1. Introduce a top-level boolean field named `maintainOrder` in the `.spec` section of the `BackupBatch` or `RestoreBatch`. If the value of this field is `false` (which is the default value), we can run the backup/restore in parallel. This will match our current behavior. If it is `true` then we will run backup/restore in the same order as the targets are specified in `members` section.
```
Comment (endrec) [inline keps/restore-batch.md:225]: I'd prefer this option, this maintains backward compatibilty.

**Comment by codejamninja** [comment]
This is great

File: keps/restore-batch.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    - name: wordpress-persistent-storage
+      mountPath: /var/www/html
+  paths:
+    - /var/www/html
+```
+
+Now, the `hostname` generation will follow the following patterns:
+
+1. For `Deployment`, `ReplicaSet`, `ReplicationController`, `DeploymentConfig` , and `AppBinding`,  the `alias` value will be used directly as the `hostname`.
+2. For `StatefulSet` the `hostname` will be generated as `$alias-$POD_INDEX` for the individual pods.
```
Comment (tamalsaha) [inline keps/restore-batch.md:58]: `hostname` was actually just `host`, right?

Comment (hossainemruz) [reply keps/restore-batch.md:58]: yes

File: keps/restore-batch.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
+# Introduce RestoreBatch + Fix BackupBatch Issue
```
Comment (tamalsaha) [inline keps/restore-batch.md:1]: Let's put this in the `enhancements` folder instead of `keps`.

File: keps/restore-batch.md (hunk 1 above)
Comment (tamalsaha) [inline keps/restore-batch.md:225]: `executionOrder`: `Parallel` (default) or `Sequential`

**Pull request merged**